src/ion/moves.rs

```diff
@@ -376,24 +205,16 @@ impl<'a, F: Function> Env<'a, F> {
                 // already in this range (hence guaranteed to have the
                 // same allocation) and if the vreg is live, add a
                 // Source half-move.
-                let mut block = self.cfginfo.insn_block[range.from.inst().index()];
+                let mut block = self.ctx.cfginfo.insn_block[range.from.inst().index()];
                 while block.is_valid() && block.index() < self.func.num_blocks() {
-                    if range.to < self.cfginfo.block_exit[block.index()].next() {
+                    if range.to < self.ctx.cfginfo.block_exit[block.index()].next() {
                         break;
                     }
                     trace!("examining block with end in range: block{}", block.index());
 
-                    match inter_block_sources.entry(block) {
-                        // If the entry is already present in the map, we'll try to prefer a
-                        // register allocation.
-                        Entry::Occupied(mut entry) => {
-                            if !entry.get().is_reg() {
-                                entry.insert(alloc);
-                            }
-                        }
-                        Entry::Vacant(entry) => {
-                            entry.insert(alloc);
-                        }
+                    let entry = &mut inter_block_sources[block.index()];
```

Comment:
Can we keep the original code here to avoid non-mechanical changes (and do a separate PR if this change is worthwhile)?